src/main/java/minicraft/entity/mob/ObsidianKnight.java

```diff
@@ -0,0 +1,344 @@
+package minicraft.entity.mob;
+
+import minicraft.core.Game;
+import minicraft.core.Updater;
+import minicraft.core.io.Localization;
+import minicraft.core.io.Sound;
+import minicraft.entity.*;
+import minicraft.entity.furniture.KnightStatue;
+import minicraft.gfx.Color;
+import minicraft.gfx.Font;
+import minicraft.gfx.Screen;
+import minicraft.gfx.SpriteLinker;
+import minicraft.item.Items;
+import minicraft.screen.AchievementsDisplay;
+
+import java.util.Random;
+
+public class ObsidianKnight extends EnemyMob {
+	private static final SpriteLinker.LinkedSprite[][][] armored = new SpriteLinker.LinkedSprite[][][] {
+		Mob.compileMobSpriteAnimations(0, 0, "obsidian_knight_armored"),
+		Mob.compileMobSpriteAnimations(0, 2, "obsidian_knight_armored"),
+		Mob.compileMobSpriteAnimations(0, 4, "obsidian_knight_armored"),
+		Mob.compileMobSpriteAnimations(0, 6, "obsidian_knight_armored")
+	};
+	private static final SpriteLinker.LinkedSprite[][][] broken = new SpriteLinker.LinkedSprite[][][] {
+		Mob.compileMobSpriteAnimations(0, 0, "obsidian_knight_broken"),
+		Mob.compileMobSpriteAnimations(0, 2, "obsidian_knight_broken"),
+		Mob.compileMobSpriteAnimations(0, 4, "obsidian_knight_broken"),
+		Mob.compileMobSpriteAnimations(0, 6, "obsidian_knight_broken")
+	};
+	public static ObsidianKnight entity = null;
+	public static boolean beaten = false; // If the boss was beaten
+	public static boolean active = false; // If the boss is active
+	public static boolean failed = false; // If the player dies
+	private static boolean phase1 = true; // If the boss is in phase 1 or not
```

Comment:
Yeah, I suggest making these booleans into an enum. None of these, except the singleton, should be static.

Reply:
Idk how to do enums, I usually use statics, I forget why. I was recently taught to how to use them properly however I can’t fix it myself currently.